### PR TITLE
Daura release

### PR DESCRIPTION
**Pull request by salvisolamartinell**
Daura implementation.

## Type of change

Please delete options that are not relevant.

- [x] New algorithm or support class.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. If you haven't added any test and it is relevant provide instructions so we can reproduce.

- [x] I have added a new test file: [E.g. `test_rf.py`]
- [x] I have added a new test case: [E.g. `RFTest.test_make_classification`]
- [x] I have tested it manually in a **local environment**.
- [x] I have tested it manually in a **supercomputer**.

Reproduce instructions:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have documented the public methods of any public class according to the guide styles. 
- [ ] I have made corresponding changes to the documentation
- [] New and existing unit tests pass locally with my changes
- [x] I have rebased my branch before trying to merge.

### CODE REVIEW

File: dislib/cluster/daura/base.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        """ Perform Daura clustering.
+
+        See:
+        lockhartlab.squarespace.com/blog/2018/1/14/clustering-with-daura-et-al
+
+        Parameters
+        ----------
+        cutoff : float
+            Distance to determine the neighbors of a sample.
+        """
```
Comment (javicid) [inline dislib/cluster/daura/base.py:23]: This should go below the class declaration.

Use a References subsection for the reference (see C-SVM for example)

File: docs/source/api-reference.rst [hunk 2]
```diff
@@ -76,6 +76,9 @@ clustering.
 :class:`cluster.GaussianMixture <dislib.cluster.gm.base.GaussianMixture>` -
 Fit a gaussian mixture model.
 
+:class:`cluster.Daura <dislib.cluster.daura.base.Daura>` -
+Fit a gaussian mixture model.
```
Comment (javicid) [inline docs/source/api-reference.rst:80]: The description is wrong

File: dislib/cluster/daura/base.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        """ Compute Daura clustering.
+
+        Parameters
+        ----------
+        distances : ds-array (n_samples, n_samples)
+            Pairwise distances between the samples.
+
+        Returns
+        -------
+        clusters : List[ List[ int ] ]
```
Comment (javicid) [inline dislib/cluster/daura/base.py:36]: Why not return a ds-array?

**Review by javicid — CHANGES_REQUESTED** [review]

File: dislib/cluster/daura/base.py (hunk 1 above)
Comment (salvisolamartinell) [reply dislib/cluster/daura/base.py]: Done. The link wasn't a reference, just an info webpage, but I have formatted it. And I have added the reference to the original paper.

File: docs/source/api-reference.rst (hunk 2 above)
Comment (salvisolamartinell) [reply docs/source/api-reference.rst]: Corrected, thank you.

File: dislib/cluster/daura/base.py (hunk 3 above)
Comment (salvisolamartinell) [reply dislib/cluster/daura/base.py:47]: I opened an issue (#311) to solve this later.

**Review by javicid — APPROVED** [review]

**Pull request merged**